source/agora/consensus/Fee.d

```diff
@@ -401,6 +401,33 @@ public class FeeManager
         } ();
     }
 
+    /***************************************************************************
+
+        Calculate the transaction fee and adjust the fee based on the
+        transaction's size measured in bytes.
+
+        The bigger the transaction size is, the smaller the the fee becomes.
+
+        Params:
+            tx = transaction for which we want to calculate the adjusted fee
+            peekUTXO = UTXO finder
+            tot_fee = adjusted fee
+
+        Returns: string describing the error, if an error happened, null otherwise
+
+    ***************************************************************************/
+
+    public string getAdjustedTXFee (in Transaction tx, scope UTXOFinder peekUTXO,
+        out Amount tot_fee) nothrow @safe
+    {
+        Amount tot_data_fee;
+
+        auto res = getTXSetFees([tx], peekUTXO, tot_fee, tot_data_fee);
```

Comment:
`getTXSetFees` should really accept a range, or you could pass `(&tx)[0 .. 1]` to avoid the needless allocation.
But let's fix this later as it might create other issues with scoping and `@safe`.

Reply:
Just using a fixed size array with length 1 would avoid allocations.